tests/Unit/model/CategoryTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Tests\Unit\model;
+
+use App\Models\Category;
+use App\Models\Product;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\Relations\HasManyThrough;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use Tests\TestCase;
+
+class CategoryTest extends TestCase
+{
+    /**
+     * A basic unit test example.
+     *
+     * @return void
+     */
+    use DatabaseTransactions;
+
+    public function testCategoryHasManyChildCategories()
+    {
+        $category = Category::factory()->create();
+        $childcategory = Category::factory()->create([
+            'parent_id' => $category->id
+        ]);
```

Comment:
ko dùng create() ntn nhé, vì factory sẽ tạo ra data trong db, Unit test phải độc lập với các thành phần còn lại mà. tương tự bên dưới. ta ko cần setup quan hệ thật để test HasMany đâu :3 
```suggestion
        $category = new Category();
```